src/marketplace/ListingsView.js

```diff
@@ -77,7 +82,17 @@ class ListingsView extends React.Component<Props, State> {
 	//This function uses the users id and gets the tanks that are active in the marketplace
 	getMarketSalesForTanks(): void  {
 		getMarketTanks(this.state.userId, sales => {
-				this.setState({itemsForSale: sales.filter(sale => !(allComponents.includes(sale.name)))}); 
+				//Once we have all of the tanks being sold add them to the array that is going to house those tanks
+				const tanksForSale: Array<Tank> = [];
+				for(const tank of sales) {
+					getTankById(tank.tankId, tankReturned => {
+						tanksForSale.push(tankReturned);
+					});
+				}
+				this.setState({
+					itemsForSale: sales.filter(sale => !(allComponents.includes(sale.name))),
+					tanksBeingShown: tanksForSale
```

Comment:
I talked to Adam about this, but this is very confusing and probably doesn't do what you want it to in some cases. This array isn't actually full yet, so you are about to call render without the stuff you need in this array, and then it will get put into the array after the fact

Reply:
The solution is to not call setstate until after the API call you are waiting for returns